Create entire frame tree in iframe processes

achristensen07 opened the pull request:
#### 93cc4767ec04fbe1027cda51ccb2ad1d47fa879f
<pre>
Create entire frame tree in iframe processes
<a href="https://bugs.webkit.org/show_bug.cgi?id=254126">https://bugs.webkit.org/show_bug.cgi?id=254126</a>

Reviewed by Chris Dumez.

This makes it so that in a subframe process, we make the entire frame tree with a RemoteFrame as the main frame.
The next step will be to broadcast frame tree changes to all processes when an iframe is made in one process.
In order to get the pixels from the iframe to still draw to the screen, we had to introduce the concept of
root frames, which are the topmost LocalFrames in a particular process, and those are the frames that have a root
layer that is sent to the UI process in a RemoteLayerTreeTransaction.  In places where we used to tell the main
frame to draw, we now need to iterate all the root layers and tell each of them to draw.  We currently only have
unit tests that have one root layer per process, but this is a step towards having the ability to have multiple,
such as if a.com has two iframes, each of which is showing b.com.

Instead of using the DrawingArea::UpdateGeometry message to update the size of iframes, we make a new message
WebPage::UpdateFrameSize for that purpose that does many of the same things, importantly calling FrameView::resize
to make iframes draw their content into Widgets that are the right size to composite into another process&apos;s content.

I remove a hack I introduced for WebFrameProxy::didFinishLoad to ignore about:blank when considering whether to send
WebFrame::DidFinishLoadInAnotherProcess.  The hack is no longer necessary.

The ProvisionalFrameProxy constructor sent WebPage::LoadRequest messages twice.  That was a mistake that wasn&apos;t making
any tests fail yet, but we only need to send it once.

WebKitTestRunner needed a null check to not crash in the test http/tests/site-isolation/basic-iframe.html because
WKBundlePageGetMainFrame returns a WebFrame wrapping a RemoteFrame.  It&apos;s fine to just early return because we aren&apos;t
getting render tree output in any site isolation tests yet, but we will likely need to revisit that change when we do.

* Source/WebCore/loader/FrameLoader.cpp:
(WebCore::FrameLoader::didReachVisuallyNonEmptyState):
* Source/WebCore/page/Frame.cpp:
(WebCore::Frame::isRootFrame const):
* Source/WebCore/page/Frame.h:
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::checkAndDispatchDidReachVisuallyNonEmptyState):
* Source/WebCore/page/Page.cpp:
(WebCore::createMainFrame):
(WebCore::Page::finalizeRenderingUpdate):
(WebCore::Page::finalizeRenderingUpdateForRootFrame):
* Source/WebCore/page/Page.h:
(WebCore::Page::rootFrames const):
(WebCore::Page::addRootFrame):
* Source/WebCore/page/scrolling/ScrollingStateNode.h:
(WebCore::LayerRepresentation::operator PlatformLayer* const):
* Source/WebCore/page/scrolling/mac/ScrollingTreeFrameScrollingNodeMac.mm:
(WebCore::ScrollingTreeFrameScrollingNodeMac::repositionScrollingLayers):
* Source/WebCore/rendering/RenderLayerCompositor.cpp:
(WebCore::RenderLayerCompositor::isMainFrameCompositor const):
* Source/WebCore/rendering/RenderObject.cpp:
(WebCore::RenderObject::~RenderObject):
* Source/WebCore/rendering/RenderView.cpp:
(WebCore::RenderView::flushAccumulatedRepaintRegion const):
* Source/WebKit/Shared/LoadParameters.cpp:
(WebKit::LoadParameters::encode const):
(WebKit::LoadParameters::decode):
* Source/WebKit/Shared/LoadParameters.h:
* Source/WebKit/Shared/WebPageCreationParameters.cpp:
(WebKit::WebPageCreationParameters::encode const):
(WebKit::WebPageCreationParameters::decode):
(WebKit::WebPageCreationParameters::SubframeProcessFrameTreeInitializationParameters::encode const):
(WebKit::WebPageCreationParameters::SubframeProcessFrameTreeInitializationParameters::decode):
* Source/WebKit/Shared/WebPageCreationParameters.h:
* Source/WebKit/UIProcess/ProvisionalFrameProxy.cpp:
(WebKit::ProvisionalFrameProxy::ProvisionalFrameProxy):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.h:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.messages.in:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxy::commitLayerTree):
(WebKit::RemoteLayerTreeDrawingAreaProxy::commitLayerTreeTransaction):
* Source/WebKit/UIProcess/WebFrameProxy.cpp:
(WebKit::WebFrameProxy::didFinishLoad):
(WebKit::WebFrameProxy::updateRemoteFrameSize):
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::attachRootGraphicsLayer):
* Source/WebKit/WebProcess/WebCoreSupport/WebFrameLoaderClient.cpp:
(WebKit::WebFrameLoaderClient::dispatchDidReachVisuallyNonEmptyState):
* Source/WebKit/WebProcess/WebPage/DrawingArea.h:
(WebKit::DrawingArea::attachToInitialRootFrame):
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeContext.h:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeContext.mm:
(WebKit::RemoteLayerTreeContext::buildTransaction):
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.h:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.mm:
(WebKit::initialRootFrame):
(WebKit::RemoteLayerTreeDrawingArea::RemoteLayerTreeDrawingArea):
(WebKit::m_updateRenderingTimer):
(WebKit::RemoteLayerTreeDrawingArea::updateRootLayers):
(WebKit::RemoteLayerTreeDrawingArea::attachViewOverlayGraphicsLayer):
(WebKit::RemoteLayerTreeDrawingArea::attachToInitialRootFrame):
(WebKit::RemoteLayerTreeDrawingArea::setRootCompositingLayer):
(WebKit::RemoteLayerTreeDrawingArea::shouldUseTiledBackingForFrameView const):
(WebKit::RemoteLayerTreeDrawingArea::updatePreferences):
(WebKit::RemoteLayerTreeDrawingArea::updateRendering):
(WebKit::RemoteLayerTreeDrawingArea::mainFrameContentSizeChanged):
* Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
(WebKit::WebFrame::createLocalSubframeHostedInAnotherProcess):
(WebKit::WebFrame::createRemoteSubframe):
(WebKit::WebFrame::coreAbstractFrame const):
(WebKit::WebFrame::didCommitLoadInAnotherProcess):
(WebKit::WebFrame::isRootFrame const):
* Source/WebKit/WebProcess/WebPage/WebFrame.h:
(WebKit::WebFrame::layerHostingContextIdentifier):
(WebKit::WebFrame::setLayerHostingContextIdentifier):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::clientForMainFrame):
(WebKit::m_appHighlightsVisible):
(WebKit::WebPage::constructFrameTree):
(WebKit::WebPage::enterAcceleratedCompositingMode):
(WebKit::WebPage::exitAcceleratedCompositingMode):
(WebKit::WebPage::loadRequest):
(WebKit::WebPage::updateFrameSize):
(WebKit::WebPage::willCommitLayerTree):
(WebKit::WebPage::SandboxExtensionTracker::beginLoad):
(WebKit::WebPage::didCommitLoad):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
(WebKit::WebPage::layerHostingContextIdentifier const): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Source/WebKit/WebProcess/WebPage/mac/TiledCoreAnimationDrawingArea.h:
* Source/WebKit/WebProcess/WebPage/mac/TiledCoreAnimationDrawingArea.mm:
(WebKit::TiledCoreAnimationDrawingArea::setRootCompositingLayer):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/SiteIsolation.mm:
(TestWebKitAPI::TEST):
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundlePage.cpp:
(WTR::InjectedBundlePage::dump):

Canonical link: <a href="https://commits.webkit.org/261930@main">https://commits.webkit.org/261930@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3c907c819e5e59aece2c0217a430b4d5618a28c5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [⏳ 🧪 style ](https://ews-build.webkit.org/#/builders/Style-EWS "Waiting in queue, processing has not started yet") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/77/builds/22454 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [⏳ 🛠 mac ](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 wpe ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 wincairo ](https://ews-build.webkit.org/#/builders/WinCairo-EWS "Waiting in queue, processing has not started yet") 
| [⏳ 🧪 bindings ](https://ews-build.webkit.org/#/builders/Bindings-Tests-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 ios-sim ](https://ews-build.webkit.org/#/builders/iOS-16-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 mac-AS-debug ](https://ews-build.webkit.org/#/builders/macOS-AppleSilicon-Ventura-Debug-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 wpe-wk2 ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | 
| [⏳ 🧪 webkitperl ](https://ews-build.webkit.org/#/builders/WebKitPerl-Tests-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 ios-wk2 ](https://ews-build.webkit.org/#/builders/iOS-16-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 api-mac ](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 gtk ](https://ews-build.webkit.org/#/builders/GTK-Build-EWS "Waiting in queue, processing has not started yet") | 
| | [⏳ 🧪 api-ios ](https://ews-build.webkit.org/#/builders/iOS-16-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 mac-wk1 ](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 gtk-wk2 ](https://ews-build.webkit.org/#/builders/GTK-Build-EWS "Waiting in queue, processing has not started yet") | 
| [⏳ 🛠 🧪 jsc ](https://ews-build.webkit.org/#/builders/JSC-Tests-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 tv ](https://ews-build.webkit.org/#/builders/tvOS-16-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 mac-wk2 ](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 api-gtk ](https://ews-build.webkit.org/#/builders/GTK-Build-EWS "Waiting in queue, processing has not started yet") | 
| | [⏳ 🛠 tv-sim ](https://ews-build.webkit.org/#/builders/tvOS-16-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 mac-AS-debug-wk2 ](https://ews-build.webkit.org/#/builders/macOS-AppleSilicon-Ventura-Debug-Build-EWS "Waiting in queue, processing has not started yet") | | 
| | [⏳ 🛠 watch ](https://ews-build.webkit.org/#/builders/watchOS-9-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 mac-wk2-stress ](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-Build-EWS "Waiting in queue, processing has not started yet") | | 
| | [⏳ 🛠 watch-sim ](https://ews-build.webkit.org/#/builders/watchOS-9-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/4553 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/page/Frame.cpp:
```diff
@@ -53,6 +54,20 @@ Frame::~Frame()
     m_windowProxy->detachFromFrame();
 }
 
+bool Frame::isRootFrame() const
```
I think it may be nice to have a comment to explain what a root frame. It wasn't super obvious to me.

achristensen07 replied:
Good idea!

cdumez commented on Source/WebKit/Shared/WebPageCreationParameters.cpp:
```diff
@@ -682,4 +674,32 @@ std::optional<WebPageCreationParameters> WebPageCreationParameters::decode(IPC::
 
+void WebPageCreationParameters::SubframeProcessFrameTreeInitializationParameters::encode(IPC::Encoder& encoder) const
+{
+    encoder << localFrameIdentifier;
```
Kind of surprised to see you using old style encoders / decoders in new code.

achristensen07 replied:
Yeah, migration will be done separately.

cdumez commented on Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
```diff
@@ -148,6 +149,42 @@ Ref<WebFrame> WebFrame::createSubframe(WebPage& page, WebFrame& parent, const At
+{
+    auto frame = create(page);
+    RELEASE_ASSERT(page.corePage());
+    RELEASE_ASSERT(parent.coreAbstractFrame());
```
Looks like coreAbstractFrame needs renaming.

achristensen07 replied:
Will rename in a future PR.  coreFrame is already taken, and renaming that to coreLocalFrame is out of the scope of this PR.

cdumez commented on Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
```diff
@@ -213,6 +250,12 @@ WebCore::RemoteFrame* WebFrame::coreRemoteFrame() const
 }
 
+// FIXME: Remove coreRemoteFrame and replace coreFrame with this.
+WebCore::Frame* WebFrame::coreAbstractFrame() const
```
We don't use the term "abstract" frame any longer.

cdumez commented on Source/WebKit/WebProcess/WebPage/WebPage.cpp:
```diff
@@ -494,6 +495,18 @@ static Vector<UserContentURLPattern> parseAndAllowAccessToCORSDisablingPatterns(
     return parsedPatterns;
 }
 
+static std::variant<UniqueRef<FrameLoaderClient>, UniqueRef<RemoteFrameClient>> clientForMainFrame(Ref<WebFrame>&& mainFrame, bool local)
```
Do we really need a variant? Can't they have a base interface?

achristensen07 replied:
We really need a variant because we need to use the index to determine which type of Frame to create.

cdumez replied:
An interface with a `isRemoteFrameClient()` virtual function would give you the same thing, no?

cdumez commented on Source/WebKit/WebProcess/WebPage/WebPage.cpp:
```diff
@@ -990,6 +1007,20 @@ WebPage::WebPage(PageIdentifier pageID, WebPageCreationParameters&& parameters)
 
+void WebPage::constructFrameTree(WebFrame& parent, WebCore::FrameIdentifier localFrameIdentifier, WebCore::LayerHostingContextIdentifier localFrameHostLayerIdentifier, const FrameTreeCreationParameters& treeCreationParameters)
+{
+    bool createLocalFrame = treeCreationParameters.frameID == localFrameIdentifier;
```
Needs a prefix, like "should"

cdumez reviewed (approved):
LGTM

achristensen07 committed:
Create entire frame tree in iframe processes
https://bugs.webkit.org/show_bug.cgi?id=254126

Reviewed by Chris Dumez.

This makes it so that in a subframe process, we make the entire frame tree with a RemoteFrame as the main frame.
The next step will be to broadcast frame tree changes to all processes when an iframe is made in one process.
In order to get the pixels from the iframe to still draw to the screen, we had to introduce the concept of
root frames, which are the topmost LocalFrames in a particular process, and those are the frames that have a root
layer that is sent to the UI process in a RemoteLayerTreeTransaction.  In places where we used to tell the main
frame to draw, we now need to iterate all the root layers and tell each of them to draw.  We currently only have
unit tests that have one root layer per process, but this is a step towards having the ability to have multiple,
such as if a.com has two iframes, each of which is showing b.com.

Instead of using the DrawingArea::UpdateGeometry message to update the size of iframes, we make a new message
WebPage::UpdateFrameSize for that purpose that does many of the same things, importantly calling FrameView::resize
to make iframes draw their content into Widgets that are the right size to composite into another process's content.

I remove a hack I introduced for WebFrameProxy::didFinishLoad to ignore about:blank when considering whether to send
WebFrame::DidFinishLoadInAnotherProcess.  The hack is no longer necessary.

The ProvisionalFrameProxy constructor sent WebPage::LoadRequest messages twice.  That was a mistake that wasn't making
any tests fail yet, but we only need to send it once.

WebKitTestRunner needed a null check to not crash in the test http/tests/site-isolation/basic-iframe.html because
WKBundlePageGetMainFrame returns a WebFrame wrapping a RemoteFrame.  It's fine to just early return because we aren't
getting render tree output in any site isolation tests yet, but we will likely need to revisit that change when we do.

* Source/WebCore/loader/FrameLoader.cpp:
(WebCore::FrameLoader::didReachVisuallyNonEmptyState):
* Source/WebCore/page/Frame.cpp:
(WebCore::Frame::isRootFrame const):
* Source/WebCore/page/Frame.h:
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::checkAndDispatchDidReachVisuallyNonEmptyState):
* Source/WebCore/page/Page.cpp:
(WebCore::createMainFrame):
(WebCore::Page::finalizeRenderingUpdate):
(WebCore::Page::finalizeRenderingUpdateForRootFrame):
* Source/WebCore/page/Page.h:
(WebCore::Page::rootFrames const):
(WebCore::Page::addRootFrame):
* Source/WebCore/page/scrolling/ScrollingStateNode.h:
(WebCore::LayerRepresentation::operator PlatformLayer* const):
* Source/WebCore/page/scrolling/mac/ScrollingTreeFrameScrollingNodeMac.mm:
(WebCore::ScrollingTreeFrameScrollingNodeMac::repositionScrollingLayers):
* Source/WebCore/rendering/RenderLayerCompositor.cpp:
(WebCore::RenderLayerCompositor::isMainFrameCompositor const):
* Source/WebCore/rendering/RenderObject.cpp:
(WebCore::RenderObject::~RenderObject):
* Source/WebCore/rendering/RenderView.cpp:
(WebCore::RenderView::flushAccumulatedRepaintRegion const):
* Source/WebKit/Shared/LoadParameters.cpp:
(WebKit::LoadParameters::encode const):
(WebKit::LoadParameters::decode):
* Source/WebKit/Shared/LoadParameters.h:
* Source/WebKit/Shared/WebPageCreationParameters.cpp:
(WebKit::WebPageCreationParameters::encode const):
(WebKit::WebPageCreationParameters::decode):
(WebKit::WebPageCreationParameters::SubframeProcessFrameTreeInitializationParameters::encode const):
(WebKit::WebPageCreationParameters::SubframeProcessFrameTreeInitializationParameters::decode):
* Source/WebKit/Shared/WebPageCreationParameters.h:
* Source/WebKit/UIProcess/ProvisionalFrameProxy.cpp:
(WebKit::ProvisionalFrameProxy::ProvisionalFrameProxy):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.h:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.messages.in:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxy::commitLayerTree):
(WebKit::RemoteLayerTreeDrawingAreaProxy::commitLayerTreeTransaction):
* Source/WebKit/UIProcess/WebFrameProxy.cpp:
(WebKit::WebFrameProxy::didFinishLoad):
(WebKit::WebFrameProxy::updateRemoteFrameSize):
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::attachRootGraphicsLayer):
* Source/WebKit/WebProcess/WebCoreSupport/WebFrameLoaderClient.cpp:
(WebKit::WebFrameLoaderClient::dispatchDidReachVisuallyNonEmptyState):
* Source/WebKit/WebProcess/WebPage/DrawingArea.h:
(WebKit::DrawingArea::attachToInitialRootFrame):
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeContext.h:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeContext.mm:
(WebKit::RemoteLayerTreeContext::buildTransaction):
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.h:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.mm:
(WebKit::initialRootFrame):
(WebKit::RemoteLayerTreeDrawingArea::RemoteLayerTreeDrawingArea):
(WebKit::m_updateRenderingTimer):
(WebKit::RemoteLayerTreeDrawingArea::updateRootLayers):
(WebKit::RemoteLayerTreeDrawingArea::attachViewOverlayGraphicsLayer):
(WebKit::RemoteLayerTreeDrawingArea::attachToInitialRootFrame):
(WebKit::RemoteLayerTreeDrawingArea::setRootCompositingLayer):
(WebKit::RemoteLayerTreeDrawingArea::shouldUseTiledBackingForFrameView const):
(WebKit::RemoteLayerTreeDrawingArea::updatePreferences):
(WebKit::RemoteLayerTreeDrawingArea::updateRendering):
(WebKit::RemoteLayerTreeDrawingArea::mainFrameContentSizeChanged):
* Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
(WebKit::WebFrame::createLocalSubframeHostedInAnotherProcess):
(WebKit::WebFrame::createRemoteSubframe):
(WebKit::WebFrame::coreAbstractFrame const):
(WebKit::WebFrame::didCommitLoadInAnotherProcess):
(WebKit::WebFrame::isRootFrame const):
* Source/WebKit/WebProcess/WebPage/WebFrame.h:
(WebKit::WebFrame::layerHostingContextIdentifier):
(WebKit::WebFrame::setLayerHostingContextIdentifier):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::clientForMainFrame):
(WebKit::m_appHighlightsVisible):
(WebKit::WebPage::constructFrameTree):
(WebKit::WebPage::enterAcceleratedCompositingMode):
(WebKit::WebPage::exitAcceleratedCompositingMode):
(WebKit::WebPage::loadRequest):
(WebKit::WebPage::updateFrameSize):
(WebKit::WebPage::willCommitLayerTree):
(WebKit::WebPage::SandboxExtensionTracker::beginLoad):
(WebKit::WebPage::didCommitLoad):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
(WebKit::WebPage::layerHostingContextIdentifier const): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Source/WebKit/WebProcess/WebPage/mac/TiledCoreAnimationDrawingArea.h:
* Source/WebKit/WebProcess/WebPage/mac/TiledCoreAnimationDrawingArea.mm:
(WebKit::TiledCoreAnimationDrawingArea::setRootCompositingLayer):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/SiteIsolation.mm:
(TestWebKitAPI::TEST):
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundlePage.cpp:
(WTR::InjectedBundlePage::dump):

Canonical link: https://commits.webkit.org/261930@main
